### PR TITLE
Better JSON-RPC error handling in MetricsProvider

### PR DESCRIPTION
**Pull request by mikeshultz**
### Description:

Improves overall error handling, and properly tracks 429s/rate limited requests.

**BIG FLASHING NOTE**: This adds a dependency to one of **my personal repositories** because it depends on a change I made to `web3-provider-engine`.  There's a PR up (https://github.com/MetaMask/web3-provider-engine/pull/315) but no idea if it'll be accepted or not.  Would be willing to transfer my branch/repo to `OriginProtocol` project if that's more ideal.

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: packages/graphql/package.json [hunk 1]
```diff
@@ -58,7 +58,7 @@
     "rlp": "^2.2.2",
     "subscriptions-transport-ws": "^0.9.16",
     "web3": "1.0.0-beta.34",
-    "web3-provider-engine": "^15.0.0"
+    "web3-provider-engine": "mikeshultz/web3-provider-engine#too-many-requests"
```
Comment (franckc) [inline packages/graphql/package.json]: I would recommend adding a TODO to replace this package with the official one once https://github.com/MetaMask/web3-provider-engine/pull/315 is accepted and merged in.

Comment (mikeshultz) [reply packages/graphql/package.json]: Would if I could.  Can't do comments in JSON, sadly.  Recommend somewhere else?

Comment (franckc) [reply packages/graphql/package.json]: ah darn... how about in README ? not ideal but can't think of a better place ?

Comment (mikeshultz) [reply packages/graphql/package.json]: `TODO.md`?  Or is that too extreme when we have things like GH issues?

Comment (franckc) [reply packages/graphql/package.json]: Actually you are right. Filing a GH issue if probably better... I'd vote for that.

Comment (mikeshultz) [reply packages/graphql/package.json]: https://github.com/OriginProtocol/origin/issues/2388

**Pull request merged**